source/simple_socketio.ts

```diff
@@ -0,0 +1,207 @@
+      return sessionId;
+    } catch (error) {
+      console.error("Error fetching session ID:", error);
+      throw error;
+    }
+  }
+  private async initializeWebSocket(): Promise<void> {
+    const sessionId = this.sessionId ?? (await this.fetchSessionId());
+    const urlWithQueryAndSession = `${this.wsUrl}/socket.io/1/websocket/${sessionId}?${this.query}`;
+    this.ws = new WebSocket(urlWithQueryAndSession);
+    // Set transport property as a public alias of the websocket
+    this.socket.transport = this.ws;
+    this.addInitialEventListeners();
+  }
+
+  private addInitialEventListeners(): void {
+    this.ws?.addEventListener("message", this.handleMessage.bind(this));
+    this.ws?.addEventListener("open", this.handleOpen.bind(this));
+    this.ws?.addEventListener("close", this.handleClose.bind(this));
+    this.ws?.addEventListener("error", this.handleError.bind(this));
+  }
+
+  private handleError(event: Event): void {
+    this.handleClose();
+    console.error("WebSocket error:", event);
+  }
+  private handleMessage(event: MessageEvent): void {
+    const [packetType, data] = event.data.split(/:::?/);
+    if (packetType === PACKET_TYPES.event) {
+      const parsedData = JSON.parse(data) as Payload;
+      const { name, args } = parsedData;
+      this.handleEvent(name, args[0]);
+      return;
+    }
+    if (packetType === PACKET_TYPES.heartbeat) {
+      // Respond to server heartbeat with a heartbeat
+      this.ws?.send(`${PACKET_TYPES.heartbeat}::`);
+      return;
+    }
+    if (packetType === PACKET_TYPES.error) {
+      // Respond to server heartbeat with a heartbeat
+      console.log("WebSocket message error: ", event);
+      this.handleClose();
+      return;
+    }
+  }
+  private handleOpen(): void {
+    this.startHeartbeat();
+    if (this.reconnectTimeout) {
+      clearTimeout(this.reconnectTimeout);
+      this.reconnectTimeout = undefined;
+    }
+    this.handleEvent("connect", {});
+    // Set connected property to true
+    this.socket.connected = true;
+  }
+
+  private handleClose(): void {
+    this.stopHeartbeat();
+    this.scheduleReconnect();
+    // Set connected property to false
+    this.socket.connected = false;
+  }
+  private handleEvent(eventName: string, eventData?: any): void {
+    this.handlers[eventName]?.forEach((callback) => callback(eventData));
+  }
+  // Setup event callbacks for a given eventName
+  public on(eventName: string, eventCallback: (eventData: any) => void): void {
+    if (!this.handlers[eventName]) {
+      this.handlers[eventName] = [];
+    }
+    this.handlers[eventName].push(eventCallback);
+  }
+  // Emit an event with the given eventName and eventData
+  public emit(eventName: string, eventData: Event["_data"]): void {
+    const payload = {
+      name: eventName,
+      args: [eventData],
+    };
+    const dataString = eventData ? `:::${JSON.stringify(payload)}` : "";
+    this.ws?.send(`${PACKET_TYPES.event}${dataString}`);
```

Comment:
it feels a bit wrong to do nothing if the ws isn't initialized. what did the previous client do?

Reply:
Don't think it was handled, the event_hub code does unsent event queueing. But I can investigate

Reply:
I've added a simple queue